src/shared/selectors/puzzle-selectors.js

```diff
@@ -0,0 +1,6 @@
+import { createSelector } from 'reselect';
+
+export const puzzle = state => state.puzzle;
+export const bannedChars = createSelector([puzzle], puzzle => puzzle.bannedChars);
```

Comment:
I guess ‘createSelector’ is redundant here

Reply:
why

Reply:
It will unnecessary memoization of  `puzzle` for `get` operation.
Flow with `createSelector` will be `call of puzzle selector` -> `compare result with memoized` -> `get bannedChars from puzzle`. 
Without `createSelector` it will be simpler, like `call puzzle` -> `get bannedChars`.

Reply:
so why don't we need to memoize a puzzle? `bannedChars` selector will be called a lot during the round, so wouldn't it be fine to get precomputed memoized value? I know property lookup on so simple object like puzzle wouldn't take so long, so why bother, but still...

Reply:
@DaQuirm we need your valuable opinion 🙏

Reply:
But do you really? Or rather, how valuable can it actually be? 😺

Reply:
What do we use this selector for anyway? To be able to create a structured selector out of it and others?

Reply:
To me memoization seems a bit redundant too, but can't we control it somehow and get a non-memoizing selector?

Reply:
You're making things more complicated asking more questions... 😞 
My opinion: having a library which tends to be a default option for `read` operations, why don't use it? 😄  All under-the-hood-memoization-magic is kinda too complicated, so I prefer not to bother about all this complexity until we'll have performance issues (and we will not)

Reply:
What's the trade-off here? Readability/maintainability vs. potential performance penalty? I'd almost always choose the former in favor of the latter. Let's not pre-optimise, especially at a cost, this won't be a case of "too late to make it performant" anyway.

Reply:
What do you think about current approach readability?

```javascript
const puzzle = state => state.puzzle;
const title = createSelector([puzzle], puzzle => puzzle.title); // 👍 for this one
const title = puzzle(state).title); // ❤️ for this one
```

cc @kirilknysh @andrei-yanovich @KantinDmitry

Reply:
@R1ZZU yes, but what about using this selector later for a structure selector [here](https://github.com/21-23/cssqd-ui/pull/31/files#diff-c72e773e0b13a01e02e2816f9b78853fR56)? Or it doesn't matter which way we write this one?

Reply:
Don't really care. I see benefits of reselect (future extensibility, etc) but also drawbacks (slow down performance, memoizing will keep unnecessary "copy" of title)

Reply:
just add emoji FFS 😆

Reply:
> Or it doesn't matter which way we write this one?

it doesn't matter

Reply:
What about `const title = state.puzzle.title`? 🐱

Reply:
Anyway, 👍

Reply:
> What about `const title = state.puzzle.title`? 🐱

if we'll rename key in store, all "deep" selectors will fail. In case of reselect we'll need to change only root selector